config/crd/bases/risingwave.risingwavelabs.com_risingwaves.yaml

```diff
@@ -909,11 +936,34 @@ spec:
                           bucket:
                             description: Bucket of the AWS S3 service.
                             type: string
+                          endpoint:
+                            description: Endpoint of the AWS (or other vendor's S3-compatible)
+                              service. Leave it empty when using AWS S3 service. You
+                              can reference the `S3_REGION` and `S3_BUCKET` in the
+                              endpoint with `$(S3_BUCKET)` and `$(S3_REGION)`, e.g.,
+                              s3.$(S3_REGION).amazonaws.com $(S3_BUCKET).s3.$(S3_REGION).amazonaws.com
+                            pattern: ^(?:https://)?(?:[^/.\s]+\.)*(?:[^/\s]+)*$
```

Comment:
Should this pattern match the bucket url? The following are also valid URLs: 

```txt 
https://
http 
httpfsd
```
Should the pattern not match the example above? Something like `https://bucket-name.s3.region-code.amazonaws.com`

Reply:
This endpoint is used for the `s3-compatible` so any valid domain (with or without the "https://" prefix) should be a valid endpoint.

Reply:
I understand. But `http` is not a valid endpoint. It is just the beginning of the URL specifying the protocol

Reply:
Have a look at the pattern in action: https://regexr.com/71hnu

Reply:
> I understand. But http is not a valid endpoint. It is just the beginning of the URL specifying the protocol

Yes, it is. If you think of it as `https://http`, it's just a server hosted on the `http` domain.

Reply:
Alright. As long as we can handle it in the end 👍